stateful-functions.md

```diff
@@ -8,38 +8,63 @@ layout: base
 
-<!-- What is Stateful Functions? -->
 <div class="row front-graphic">
   <img src="{{ site.baseurl }}/img/stateful-functions/statefun-overview.png" width="650px"/>
 </div>
 
-The API is based on **functions with persistent state** that can share a pool of resources and **interact arbitrarily with strong consistency guarantees**.
-
 <div class="jumbotron">
-  <h2>The Building Blocks</h2>
-  <p style="font-size:110%;">A stateful function is a <span class="text-info">small piece of logic/code</span> that can be invoked through a message and is:</p>
-    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Virtual</b> Functions exist as virtual instances with a logical address.</p>
-    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Addressable</b> Functions interact with each other by sending point-to-point messages to these addresses.</p>
-    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Stateful</b> Functions have locally embedded, fault-tolerant state provided and managed by Apache Flink.</p>
-    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Lightweight</b> Functions don't consume resources unless actively being invoked.</p>
-  <p style="font-size:110%;">Applications are <span class="text-info">composed from multiple functions that message each other in arbitrary, potentially cyclic ways</span> .</p>
+  <p style="font-size:110%;">A stateful function is a <span class="text-info">small piece of logic/code</span>, existing in many instances that represent entities, similar to actors. The functions is invoked through a message and are:</p>
+ 
+    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Stateful</b> Functions have embedded, fault-tolerant state, accessed locally like a variable.</p>
+    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Virtual</b> Like FaaS, stateful functions don't reserve resources. Inactive functions don't consume CPU/Memory.</p>
+  
+  <p style="font-size:110%;">Applications are <span class="text-info">composed from multiple functions</span> that <span class="text-info">message each other</span>:</p>
+    
+    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Exactly-once</b> State and messaging go hand-in-hand, providing exactly once message/state semantics.</p>
+    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Logical Addressing</b> Functions message each other by logical addresses. No name resulution needed.</p>
+    <p style="font-size:100%;"><span class="glyphicon glyphicon glyphicon-check"></span><b> Cyclic data flows</b> In contrast to stream processing, messaging patterns are not restricted to DAGs.</p>
```

Comment:
Why is this relevant from a user's perspective? Isn't the cyclic data flow how we implement statefun on top of Flink? The point below "dynamic messaging" is a much stronger argument.

Reply:
Its is very important, because it allows you to implement programs that you just cannot implement in stream processing.

Reply:
But it is a bit confusing to list "Cyclic **data flows**" as a benefit, while the next benefit says "Messaging patterns are not required to be pre-defined in a dataflow". 
Maybe we can merge "Cyclic data flows" and "Dynamic messaging" into one item?

But I'm probably a bit pedantic here :) Please don't consider my opinion as a strong point.

Reply:
I see. Maybe we can call it "cyclic message flows" or so? This is a differentiator to all other stream processing systems, so it would be nice to have that message somewhere in there.

Reply:
How about "Cyclic messaging"? (to get rid of the "flow")

I just don't want this to sound too much like a stream processor, rather like a generic application development framework